File: 2017/01/inverse_captcha.go
package main

import (
	"fmt"
	"io/ioutil"
	"strconv"
)

func main() {
	dat, _ := ioutil.ReadFile("input")

	prev := dat[len(dat)-1]
	total := 0
	for _, item := range dat {
		if item == prev {
			i, _ := strconv.Atoi(string(item))
			total += i
		}
		prev = item
	}
	fmt.Println(total)

	halfway := len(dat) / 2

	total = 0
	for i := range dat {
		if dat[i] == dat[(i+halfway)%len(dat)] {
			j, _ := strconv.Atoi(string(dat[i]))
			total += j
		}
	}
	fmt.Println(total)

}

File: 2017/01/inverse_captcha.py
#env python3

with open("input") as f:
    input_string = f.read()


prev = input_string[-1]
total = 0
for digit in input_string:
    if digit == prev:
        total += int(digit)
    prev = digit
print(total)

halfway = int(len(input_string) / 2)

total = 0
for i in range(len(input_string)):
    if input_string[i] == input_string[(i + halfway) % len(input_string)]:
        total += int(input_string[i])
print(total)